src/flow-hash.c

```diff
@@ -486,6 +486,11 @@ static Flow *FlowGetNew(ThreadVars *tv, DecodeThreadVars *dtv, const Packet *p)
 
             f = FlowGetUsedFlow(tv, dtv);
             if (f == NULL) {
+                /* max memcap reached, so increments the counter */
+                if (tv != NULL && dtv != NULL) {
+                    StatsIncr(tv, dtv->counter_flow_memcap);
```

Comment:
You will also need to add this to like 503 below. Due to threading we can have a situation where the memcap wasn't reached yet at like 476, but it is reached when we enter FlowAlloc in like 501. Another thread may have filled the last available 'memcap space' in between.